src/components/product/ProductGrid.tsx

```diff
@@ -30,7 +30,7 @@ export const ProductGrid: React.FC = () => {
     return (
         <div
             data-testid='product-grid'
-            className='flex flex-wrap justify-center gap-4 md:justify-start'
+            className='grid grid-cols-1 justify-center gap-6 sm:grid-cols-2 md:justify-start lg:grid-cols-3 xxl:grid-cols-4'
```

Comment:
This is great touch, but pls add a max-width to the product cards. 
<img width="1429" alt="Screenshot 2024-10-10 at 21 52 57" src="https://github.com/user-attachments/assets/9a3a2b6c-b1a9-4404-8d10-ddd2295357a3">